packages/oscal-react-library/src/components/OSCALCatalogBaseline.tsx

```diff
@@ -0,0 +1,1518 @@
+  const [AddAuthorDetails, setAddAuthorDetails] = useState(false);
+  const [catalogIds, setCatalogIds] = useState<string[]>([]);
+  const [baselineIds, setBaselineIds] = useState<string[]>([]);
+  const Data: CatalogBaseline = {
+    isCatalog: true,
+    isVisible: true,
+  };
+  const [mainData, setMainData] = useState<CatalogBaseline>(Data);
+  const [orgAddress, setOrgAddress] = useState<Address>({});
+  const [orgContact, setOrgContact] = useState<ContactInfo>({});
+  const [authorAddress, setAuthorAddress] = useState<Address>({});
+  const [authorContact, setAuthorContact] = useState<ContactInfo>({});
+  const [openCatalogBaseline, setOpenCatalogBaseline] = useState(false);
+  const [newOSCALModel, setNewOSCALModel] = useState<CatalogBaseline | undefined>(undefined);
+  let newModelCreationDone = false;
+  let address: Address = orgAddress;
+  let contact: ContactInfo = {};
+  address = authorAddress;
+  const createdModel = newOSCALModel === undefined ? Data : newOSCALModel;
+  const LabelText = openCatalogBaseline ? createdModel.title : " Catalogs & Baselines";
+
+  useEffect(() => {
+    getCatalogIds();
+    getBaselineIds();
+  }, []);
+
+  function getCatalogIds() {
+    const request_json = {
+      method: "GET",
+    };
+    fetchRest(
+      "/catalog",
+      request_json,
+      getCatalogProjectsStatusSuccess,
+      getCatalogProjectsStatusFail
+    );
+
+    function getCatalogProjectsStatusSuccess(response: {
+      [x: string]: React.SetStateAction<string[]>;
+    }) {
+      setCatalogIds(response["projects"]);
+    }
+    function getCatalogProjectsStatusFail(e: any) {
+      console.log("Operation fail " + e.statusText);
+    }
+  }
+
+  function renderTabs() {
+    if (openCatalogBaseline) return <CatalogBaselineTabs></CatalogBaselineTabs>;
+    else return null;
+  }
+
+  function getBaselineIds() {
+    const request_json = {
+      method: "GET",
+    };
+    fetchRest(
+      "/profile",
+      request_json,
+      getProfileProjectsStatusSuccess,
+      getProfileProjectsStatusFail
+    );
+
+    function getProfileProjectsStatusSuccess(response: {
+      [x: string]: React.SetStateAction<string[]>;
+    }) {
+      setBaselineIds(response["projects"]);
+    }
+    function getProfileProjectsStatusFail(e: any) {
+      console.log("Operation fail " + e.statusText);
+    }
+  }
+  class DeleteCatalogBaseline extends React.Component {
+    render() {
+      return (
+        <Hug>
+          <Grid spacing={1}>
+            <OSCALUnsatisfiedButton sx={{ width: 57, height: 20, color: "#FFD9D9" }}>
+              DELETE
+            </OSCALUnsatisfiedButton>
+            <OSCALSecondaryButton
+              sx={{ width: 88, height: 20 }}
+              onClick={handleAddNewCatalogBaseline}
+            >
+              PUBLISH
+            </OSCALSecondaryButton>
+          </Grid>
+        </Hug>
+      );
+    }
+  }
+  class HugHug extends React.Component {
```

Comment:
Two things here:
1) We haven't been using `class` throughout the application anywhere else except for some of the type objects Kyle had created. To make things consistent could we turn all `class`es into `function`s or `const`s for component rendering code like this? We can remove the `extends React.Component` part and `render()` function this way.
2) A bit of a nitpick, but could we name this something a little more description than "HugHug", maybe something focused around the `Create New` button handling a new catalog or baseline.